File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/CS/Cliente/OpcionSalir.cs
using System;
using System.Collections.Generic;
using System.Text;

using DCE05.Ejemplos.EstrellaUno.ReglasNegocio;

namespace DCE05.Ejemplos.EstrellaUno.Cliente {

    /// <summary>
    /// Representa la opción de salir del sistema y finalizar la venta.
    /// </summary>
    internal class OpcionSalir : Opcion {

        /// <summary>
        /// Ejecuta la acción asociada a la opción.
        /// </summary>
        internal override void EjecutarAccion() {
            try {
                CatalogoVentas catalogo = new CatalogoVentas();
                if (PuntoDeVenta.VentaActual != null) {
                    catalogo.CancelarVenta(PuntoDeVenta.VentaActual);
                }
                catalogo.CancelarVentasPendientes();
                PuntoDeVenta.Salir();
            } catch (ReglasNegocioException ex) {
                Console.WriteLine("Error al cancelar las ventas : " + ex.Message);
            }
        }

        /// <summary>
        /// Construye una instancia de la opción con sus datos básicos.
        /// </summary>
        /// <param name="codigo">El código de la opción.</param>
        internal OpcionSalir(int codigo) {
            Codigo = codigo;
            Descripcion = "Salir";
        }

    }
}

File: UTESA/DCE 5 Estrellas/dce2005/Estrella1/DCE1_Ejemplos/src/VB/Cliente/OpcionSalir.vb
Imports DCE05.Ejemplos.EstrellaUno.ReglasNegocio

Public Class OpcionSalir
    Inherits Opcion

    ''' <summary>
    ''' Ejecuta la acción asociada a la opción.
    ''' </summary>
    Friend Overrides Sub EjecutarAccion()
        Try
            Dim Catalogo As CatalogoVentas = New CatalogoVentas()
            If Not PuntoDeVenta.VentaActual Is Nothing Then
                Catalogo.CancelarVenta(PuntoDeVenta.VentaActual)
            End If
            Catalogo.CancelarVentasPendientes()
            PuntoDeVenta.Salir()
        Catch Ex As ReglasNegocioException
            Console.WriteLine("Error al cancelar las ventas : " + Ex.Message)
        End Try
    End Sub

    ''' <summary>
    ''' Construye una instancia de la opción con sus datos básicos.
    ''' </summary>
    ''' <param name="MiCodigo">El código de la opción.</param>
    Friend Sub New(ByVal MiCodigo As Integer)
        Me.Codigo = MiCodigo
        Descripcion = "Salir"
    End Sub

End Class
